### ISSUE TITLE
Passing unique identifier from external provider to the client

### ISSUE THREAD

**Issue by HansKindberg** [IdentityServer/IdentityServer4#4350]
Hi

I am working on an IdentityServer4 solution with only external providers. **Do I still need to set my own unique "sub"-claim for the token passed to the client? Or can I use the unique identifier I get from the external provider?**

I have read here, that it is not best practice to pass the unique identifier from the external provider to the client:

https://stackoverflow.com/a/52669623/3588375

_However, it may not be a best practice to pass the unique identification of users (subject claim or another claim unique identifying the user) from the external provider to your client applications. You may want to map that user information with your own unique identification of the user. In that case you would need a persistent user store to hold on to your own unique user identification._

**In other words, can I skip the AutoProvisionUser-part and create an IdentityServerUser with the data/claims I get from the external provider?**

https://github.com/IdentityServer/IdentityServer4/blob/master/src/IdentityServer4/host/Quickstart/Account/ExternalController.cs#L96

Regards Hans

**Comment by leastprivilege** [comment]
https://youtu.be/QpkVnB-N20c?t=1420

**Comment by HansKindberg** [comment]
Thank you very much Dominick
